internal/security/bootstrapper/mosquitto/handlers/handlers_test.go

```diff
@@ -0,0 +1,249 @@
+* in compliance with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed under the License
+* is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+* or implied. See the License for the specific language governing permissions and limitations under
+* the License.
+*
+*******************************************************************************/
+
+package handlers
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/mosquitto/config"
+	messagebus "github.com/edgexfoundry/edgex-go/internal/security/bootstrapper/mosquitto/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/interfaces/mocks"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/messaging"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+)
+
+type Testdata struct {
+	lc         logger.LoggingClient
+	dic        *di.Container
+	secretData map[string]string
+	ctx        context.Context
+}
+
+func setUp(t *testing.T, secretName string, brokerConfigFile string, passwordFile string) *Testdata {
+
+	mockLc := logger.NewMockClient()
+	configuration := &config.ConfigurationStruct{
+		MessageQueue: config.MessageQueueInfo{
+			SecretName:       secretName,
+			BrokerConfigFile: brokerConfigFile,
+			PasswordFile:     passwordFile,
+		},
+	}
+
+	return &Testdata{
+		lc: mockLc,
+		dic: di.NewContainer(di.ServiceConstructorMap{
+			container.LoggingClientInterfaceName: func(get di.Get) interface{} {
+				return mockLc
+			},
+			messagebus.ConfigurationName: func(get di.Get) interface{} {
+				return configuration
+			},
+		}),
+		secretData: map[string]string{
+			messaging.SecretUsernameKey: "username",
+			messaging.SecretPasswordKey: "password",
+		},
+		ctx: context.Background(),
+	}
+}
+func TestHandler_GetCredentials(t *testing.T) {
+
+	// setup mock secret client
+	expectedSecretData := map[string]string{
+		"username": "TEST_USER",
+		"password": "TEST_PASS",
+	}
+
+	mockSecretProvider := &mocks.SecretProvider{}
+	mockSecretProvider.On("GetSecret", "mqtt-bus").Return(expectedSecretData, nil)
+
+	mockSecretProvider.On("GetSecret", "").Return(nil, errors.New("Empty Secret Name"))
+	mockSecretProvider.On("GetSecret", "notfound").Return(nil, errors.New("Not Found"))
+
+	type args struct {
+		ctx          context.Context
+		startupTimer startup.Timer
+		dic          *di.Container
+	}
+	tests := []struct {
+		name       string
+		args       args
+		secretName string
+		want       bool
+	}{
+		{
+			name: "GetCredentials ok",
+			args: args{
+
+				startupTimer: startup.NewTimer(3, 1),
+			},
```

Comment:
This is the same in all test case, so isn't needed as test case definition here. Just use in the test execution each time.